jablib/src/main/java/org/jabref/logic/crawler/AutomaticDuplicateRemover.java

```diff
@@ -0,0 +1,53 @@
+package org.jabref.logic.crawler;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jabref.logic.database.DuplicateCheck;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.BibEntryTypesManager;
+
+public class AutomaticDuplicateRemover {
+    private final BibEntryTypesManager bibEntryTypesManager;
+
+    public AutomaticDuplicateRemover(BibEntryTypesManager bibEntryTypesManager) {
+        this.bibEntryTypesManager = bibEntryTypesManager;
+    }
+
+    public void removeDuplicates(BibDatabaseContext databaseContext) {
+        DuplicateCheck duplicateCheck = new DuplicateCheck(bibEntryTypesManager);
+        BibDatabase database = databaseContext.getDatabase();
+        List<BibEntry> entries = database.getEntries();
+        List<BibEntry> entriesToRemove = new ArrayList<>();
+        Set<BibEntry> handledEntries = new HashSet<>();
```

Comment:
The way the looping is done now, it is guaranteed that you won't "handle" the same pair of entries twice, so the handledEntries set and the conditions are redundant.